MatrixKit/Controllers/MXKCallViewController.m

```diff
@@ -808,51 +822,58 @@ - (IBAction)onButtonPressed:(id)sender
 
 - (void)showAudioDeviceOptions
 {
-    //  create the alert
-    currentAlert = [UIAlertController alertControllerWithTitle:nil
-                                                       message:nil
-                                                preferredStyle:UIAlertControllerStyleActionSheet];
-    
-    MXWeakify(self);
+    NSMutableArray<UIAlertAction *> *actions = [NSMutableArray new];
+    NSArray<MXAudioOutputRoute *> *availableRoutes = mxCall.audioOutputRouter.availableOutputRoutes;
     
-    //  headset action
-    UIAlertAction *headsetAction = [UIAlertAction actionWithTitle:[NSBundle mxk_localizedStringForKey:@"call_more_actions_audio_use_headset"]
-                                                                style:UIAlertActionStyleDefault
-                                                              handler:^(UIAlertAction * action) {
-        
-        MXStrongifyAndReturnIfNil(self);
-        self->currentAlert = nil;
-        self->mxCall.audioToSpeaker = NO;
+    for (MXAudioOutputRoute *route in availableRoutes)
+    {
+        //  route action
+        NSString *name = route.name;
+        if (route.routeType == MXAudioOutputRouteTypeLoudSpeakers)
+        {
+            name = [NSBundle mxk_localizedStringForKey:@"call_more_actions_audio_use_device"];
```

Comment:
I wonder if we should have a localisation string for every route type so that the UI will not depend on the system and it will be more robust and predictive.

Reply:
answer: no. `route.name` can be the real device name. This is useless to try to localise it...